compiler/annotation/type_seeder.rs

```diff
@@ -625,17 +625,12 @@ impl UnaryConstraint for Kind<Variable> {
 impl UnaryConstraint for Label<Variable> {
     fn apply<Snapshot: ReadableSnapshot>(
         &self,
-        seeder: &TypeGraphSeedingContext<'_, Snapshot>,
+        _seeder: &TypeGraphSeedingContext<'_, Snapshot>,
         graph_vertices: &mut VertexAnnotations,
     ) -> Result<(), TypeInferenceError> {
-        let annotation_opt = get_type_annotation_from_label(
-            seeder.snapshot,
-            seeder.type_manager,
-            &encoding::value::label::Label::parse_from(self.type_label()),
-        )
-        .map_err(|source| TypeInferenceError::ConceptRead { source })?;
+        let annotation_opt = graph_vertices.get(self.type_label());
```

Comment:
This feels like it relies on `self.type_label()` being processed first. Using `get_type_annotation_from_label` feels safer.